editor/src/components/canvas/canvas-strategies/absolute-move-strategy.ts

```diff
@@ -38,27 +45,58 @@ export const absoluteMoveStrategy: CanvasStrategy = {
       interactionState.interactionData.drag != null
     ) {
       const drag = interactionState.interactionData.drag
-      return canvasState.selectedElements.flatMap((selectedElement) => [
-        adjustNumberProperty(
-          'permanent',
-          selectedElement,
-          stylePropPathMappingFn('left', ['style']),
-          drag.x,
-          true,
-        ),
-        adjustNumberProperty(
-          'permanent',
-          selectedElement,
-          stylePropPathMappingFn('top', ['style']),
-          drag.y,
-          true,
-        ),
+      return [
+        ...canvasState.selectedElements.flatMap((selectedElement) => {
+          const element: JSXElement | null = withUnderlyingTarget(
+            selectedElement,
+            canvasState.projectContents,
+            {},
+            canvasState.openFile,
+            null,
+            (_, e) => e,
+          )
+          const elementParentBounds =
+            MetadataUtils.findElementByElementPath(
+              sessionState.startingMetadata, // TODO should this be using the current metadata?
+              selectedElement,
+            )?.specialSizeMeasurements.immediateParentBounds ?? null
+
+          if (element == null) {
+            return []
+          }
+
+          return mapDropNulls(
+            (pin) => {
+              const horizontal = isHorizontalPoint(
+                // TODO avoid using the loaded FramePoint enum
+                framePointForPinnedProp(pin),
+              )
+              const negative = pin === 'right' || pin === 'bottom'
+              const value = getLayoutProperty(pin, right(element.props), ['style'])
+              if (isRight(value) && value.value != null) {
+                // TODO what to do about missing properties?
+
+                return adjustCssLengthProperty(
+                  'permanent',
+                  selectedElement,
+                  stylePropPathMappingFn(pin, ['style']),
+                  (horizontal ? drag.x : drag.y) * (negative ? -1 : 1),
+                  horizontal ? elementParentBounds?.width : elementParentBounds?.height,
+                  true,
+                )
+              } else {
+                return null
+              }
+            },
+            ['top', 'bottom', 'left', 'right'] as const,
+          )
+        }),
         wildcardPatch('transient', {
```

Comment:
I know this is not new code, but this should have its own `setHighlightedViews` command

Reply:
I made a todo in paper